### PR TITLE
license

### PR DESCRIPTION
**Pull request by ErnestaP**
The license can be taken from: 
license-type attribute + url (take the version)
if the license-type is not present, extract type and version from url 
if the license-type and url are not present extract from license text

The license can be just CC-BY, on any other it should crash

### CODE REVIEW

File: dags/iop/parser.py [hunk 1]
```diff
@@ -183,3 +188,25 @@ def _get_country(self, article, id):
...
+            try:
+                url = license_node.attrib["{http://www.w3.org/1999/xlink}href"]
+                match = LICENSE_PATTERN.match(url)
+                version = match.group(2)
+                license_type = match.group(1)
+                license = construct_license(url, license_type.upper(), version)
+                if license:
+                    licenses.append(license)
+            except (AttributeError, KeyError):
+                self.logger.error("License URL is not found in XML")
```
Comment (MJedr) [inline dags/iop/parser.py]: better add doi to the msg

File: dags/iop/parser.py [hunk 2]
```diff
@@ -183,3 +188,25 @@ def _get_country(self, article, id):
...
+                try:
+                    license_statement = license_node.find("license-p/ext-link").text
+                    if license_statement:
+                        licenses.append(license_statement)
+                except AttributeError:
+                    self.logger.error("License type and version is not found in XML")
```
Comment (MJedr) [inline dags/iop/parser.py]: same

File: tests/units/iop/data/no_license_statement.xml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            </abstract>
+            <custom-meta-group>
+                <custom-meta xlink:type="simple">
+                    <meta-name>arxivppt</meta-name>
+                    <meta-value>2108.04010</meta-value>
+                </custom-meta>
+            </custom-meta-group>
+        </article-meta>
+    </front>
+</article>
```
Comment (MJedr) [inline tests/units/iop/data/no_license_statement.xml]: new line needed :)

File: tests/units/iop/data/no_license_and_no_statement.xml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            </abstract>
+            <custom-meta-group>
+                <custom-meta xlink:type="simple">
+                    <meta-name>arxivppt</meta-name>
+                    <meta-value>2108.04010</meta-value>
+                </custom-meta>
+            </custom-meta-group>
+        </article-meta>
+    </front>
+</article>
```
Comment (MJedr) [inline tests/units/iop/data/no_license_and_no_statement.xml]: new line

File: .github/workflows/test-and-build.yml [hunk 5]
```diff
@@ -69,11 +69,12 @@ jobs:
...
-          client-payload: '{"project": "scoap3", "application":"scoap3-workflows", "namespace": "scoap3-qa", "image":"registry.cern.ch/cern-sis/workflows","new_tag":"${{ github.sha }}"}'
+  Deploy:
+    needs: [Build]
+    uses: cern-sis/gh-workflows/.github/workflows/deploy-qa.yml@main
+    secrets: inherit
+    with:
+      project: scoap3
+      application: scoap3-workflows
+      namespace: scoap3-qa
+      image: registry.cern.ch/cern-sis/workflows
```
Comment (ErnestaP) [inline .github/workflows/test-and-build.yml]: Changes from Jimil, I just merged it from the main branch

**Review by MJedr — APPROVED** [review]

**Comment by MJedr** [comment]
I don't know what is your git workflow, but can you maybe rebase master instead of merging it to your pr?

File: dags/common/utils.py [hunk 6]
```diff
@@ -67,3 +61,45 @@ def extract_text(article, path, field_name, dois):
...
+
+def get_license_type(license_text):
+    if CREATIVE_COMMONS_PATTERN.search(license_text):
+        first_part_of_type = "CC"
+    else:
+        raise UnknownLicense(license=license_text)
+    if BY_PATTERN.match(license_text):
+        second_part_of_type = "BY"
+        return "-".join([first_part_of_type, second_part_of_type])
+    raise UnknownLicense(license=license_text)
```
Comment (drjova) [inline dags/common/utils.py]:
```suggestion
def get_license_type(license_text):
    if not CREATIVE_COMMONS_PATTERN.search(license_text) or not BY_PATTERN.match(license_text):
        raise UnknownLicense(license=license_text)
    return "CC-BY"
```

File: dags/iop/parser.py [hunk 7]
```diff
@@ -267,3 +282,55 @@ def _get_copyright(self, article):
...
+                    "License URL is not found in XML or unknown license found. Parsing from license text..."
+                )
+                type_and_version = self._get_license_from_text(
+                    license_node=license_node
+                )
+                if type_and_version:
+                    licenses.append(type_and_version)
+                else:
+                    self.logger.error("License is not found in license text.")
+        return licenses
```
Comment (drjova) [inline dags/iop/parser.py]: @ErnestaP could you please write each case of licenses? I think this is too complicated for what we actually need.

Comment (drjova) [reply dags/iop/parser.py]: Could you please also give me some examples from the actual records?

**Pull request merged**